flow-server/src/main/java/com/vaadin/flow/router/internal/AbstractRouteRegistry.java

```diff
@@ -196,6 +211,80 @@ public List<RouteData> getRegisteredRoutes() {
         return getRegisteredRoutes(getConfiguration());
     }
 
+    @Override
+    public List<RouteData> getRegisteredAccessibleMenuRoutes(
+            VaadinRequest vaadinRequest,
+            Collection<BeforeEnterListener> accessControls) {
+        final VaadinService vaadinService = VaadinService.getCurrent();
```

Comment:
Could it be taken from `VaadinRequest`?
```suggestion
        final VaadinService vaadinService = vaadinRequest.getService();
```